### PR TITLE
refactor(strings): move string API to types module

### PR DESCRIPTION
**Pull request by cpcloud**
This PR moves string operations to their respective types/ module.

### CODE REVIEW

File: ibis/expr/types/strings.py [hunk 1]
```diff
@@ -1,18 +1,836 @@
...
+        ----------
+        substr
+            Substring for which to check
+
+        Returns
+        -------
+        BooleanValue
+            Boolean indicating the presence of `substr` in the expression
+        """
+        return self.find(substr) >= 0
```
Comment (kszucs) [inline ibis/expr/types/strings.py:182]: Eventually if there are backends with specific support for this we may want to introduce a `StringContains` operator and implement this for backends not supporting it.

**Review by kszucs — APPROVED** [review]
LGTM

File: ibis/expr/types/strings.py (hunk 1 above)
Comment (cpcloud) [reply ibis/expr/types/strings.py:182]: Agreed. It looks like DuckDB has `instr` and `position(needle in haystack)` for this.

Comment (kszucs) [reply ibis/expr/types/strings.py:182]: Can you create a follow-up ticket?

**Pull request merged**